api/api.go

```diff
@@ -328,84 +335,86 @@ func (a *API) SingleProfileQuery(r *http.Request) (*profile.Profile, *ApiError)
 	)
 }
 
-func (a *API) profileByParameters(ctx context.Context, mode, time, query, from, to string) (*profile.Profile, *ApiError) {
+func (a *API) profileByParameters(ctx context.Context, mode, time, query, from, to string) (*profile.Profile, storage.Warnings, *ApiError) {
 	switch mode {
 	case "merge":
 		f, err := parseTime(from)
 		if err != nil {
-			return nil, &ApiError{Typ: ErrorBadData, Err: err}
+			return nil, nil, &ApiError{Typ: ErrorBadData, Err: err}
 		}
 
 		t, err := parseTime(to)
 		if err != nil {
-			return nil, &ApiError{Typ: ErrorBadData, Err: err}
+			return nil, nil, &ApiError{Typ: ErrorBadData, Err: err}
 		}
 
 		if t.Before(f) {
 			err := errors.New("to timestamp must not be before from time")
-			return nil, &ApiError{Typ: ErrorBadData, Err: err}
+			return nil, nil, &ApiError{Typ: ErrorBadData, Err: err}
 		}
 
 		sel, err := parser.ParseMetricSelector(query)
 		if err != nil {
-			return nil, &ApiError{Typ: ErrorBadData, Err: err}
+			return nil, nil, &ApiError{Typ: ErrorBadData, Err: err}
 		}
 
 		return a.mergeProfiles(ctx, f, t, sel)
 	case "single":
 		t, err := parseTime(time)
 		if err != nil {
 			err = fmt.Errorf("unable to parse time: %w", err)
-			return nil, &ApiError{Typ: ErrorBadData, Err: err}
+			return nil, nil, &ApiError{Typ: ErrorBadData, Err: err}
 		}
 
 		sel, err := parser.ParseMetricSelector(query)
 		if err != nil {
 			err = fmt.Errorf("unable to parse query: %w", err)
-			return nil, &ApiError{Typ: ErrorBadData, Err: err}
+			return nil, nil, &ApiError{Typ: ErrorBadData, Err: err}
 		}
 
 		profile, err := a.findProfile(ctx, t, sel)
 		// TODO(bwplotka): Handle warnings.
 		if err != nil {
 			err = fmt.Errorf("unable to find profile: %w", err)
-			return nil, &ApiError{Typ: ErrorInternal, Err: err}
+			return nil, nil, &ApiError{Typ: ErrorInternal, Err: err}
 		}
 		if profile == nil {
-			return nil, &ApiError{Typ: ErrorNotFound, Err: errors.New("profile not found")}
+			return nil, nil, &ApiError{Typ: ErrorNotFound, Err: errors.New("profile not found")}
 		}
 
-		return profile, nil
+		return profile, nil, nil
 	default:
-		return nil, &ApiError{Typ: ErrorBadData, Err: errors.New("no mode specified")}
+		return nil, nil, &ApiError{Typ: ErrorBadData, Err: errors.New("no mode specified")}
 	}
 }
 
-func (a *API) DiffProfiles(r *http.Request) (*profile.Profile, *ApiError) {
+func (a *API) DiffProfiles(r *http.Request) (*profile.Profile, storage.Warnings, *ApiError) {
 	ctx := r.Context()
 
-	profileA, apiErr := a.profileByParameters(ctx,
+	profileA, warnings1, apiErr := a.profileByParameters(ctx,
```

Comment:
nitpicking: Maybe for consistency 
```suggestion
	profileA, warningsA, apiErr := a.profileByParameters(ctx,
```
Same for `s/warnings2/warningsB/`

Reply:
whops, yes agreed